src/requirements/courses-json-generator.ts

```diff
@@ -0,0 +1,26 @@
+import { writeFileSync } from 'fs';
+import filteredAllCourses from './filtered-all-courses';
+
+type CourseJson = {
+  [courseCode: string]: {
+    i: number; // course id
+    t: string; // title
+    r: string; // roster (aka semester)
```

Comment:
following the convention of the original json: use single letter field name to keep the json small